boto/utils.py

```diff
@@ -860,11 +864,12 @@ def notify(subject, body=None, html_body=None, to_string=None,
 
 
 def get_utf8_value(value):
-    if not six.PY2 and isinstance(value, bytes):
+    if six.PY3:
+        if isinstance(value, bytes):
+            return value.decode('utf-8')
```

Comment:
Why is this being decoded into unicode? That's not what this method's name indicates that it does.

Did you mean to just write `value.decode('utf-8')`, without the return statement, so that this fails if the bytes can't be encoded using the utf-8 codec?

Reply:
I honestly don't know the rationale, this was pulled in from Ed's changes. Does changing line 869 to `value = value.decode('utf-8')` sound reasonable? I think you're right, the return statement looks like it was probably a typo or something.

Reply:
Updated to `value = value.decode('utf-8')` per above in most recent commit.

Reply:
My issue with this change is that this method is called "get_utf8_value", meaning that the thing being returned should be a utf-8-encoded `bytes` object, not a unicode object.  The only reason that a `value.decode('utf-8')` line would be present would be to make sure that the bytes passed in could be decoded using UTF-8.

Given this confusion, I think it's worth also adding a comment to the docstring saying that the return type of this method is a `bytes` value.